packages/core/src/scripts/migrate-oz-cli-project.ts

```diff
@@ -0,0 +1,426 @@
+import path from 'path';
+import chalk from 'chalk';
+import { promises as fs } from 'fs';
+import { ManifestData, ImplDeployment } from '../manifest';
+import type { StorageItem, StorageLayout, TypeItem, TypeMembers, StructMember } from '../storage';
+
+const OPEN_ZEPPELIN_FOLDER = '.openzeppelin';
+const EXPORT_FILE = 'openzeppelin-cli-export.json';
+const PROJECT_FILE = path.join(OPEN_ZEPPELIN_FOLDER, 'project.json');
+const SUCESS_CHECK = chalk.keyword('green')('✔') + ' ';
+
+export async function migrateLegacyProject(): Promise<void> {
+  const manifestsMigrationData = await migrateManifestFiles();
+
+  const { compiler } = await getProjectFile();
+  const exportData = {
+    networkFiles: manifestsMigrationData,
+    compiler,
+  };
+
+  await writeJSONFile(EXPORT_FILE, exportData);
+  console.log(SUCESS_CHECK + `Migration data exported to ${EXPORT_FILE}`);
+  await deleteLegacyFiles();
+
+  console.log("\nThese were your project's compiler options:");
+  console.log(compiler);
+}
+
+async function migrateManifestFiles() {
```

Comment:
I want to propose a refactor here. First of all, the names `migrateManifestFiles` and `migrateManifests` are too similar. Secondly, I find the separation of concerns to be kind of weird, since one function reads from the file system and the other writes to it; I would expect those two things to happen in the same function/abstraction layer.

With that in mind I would rename the second function to `migrateManifestsData` and make it so it receives the legacy manifest objects already read from disk. But I would like to know if you have any other ideas for refactoring.

I find there are too many layers of "migration" and it becomes hard to follow.

Reply:
I separated the read + migration functions from the write ones to make sure the migration is successful before rewriting any file. The idea is to fail before touching anything.

The migration of the files depends on the successful migration of the abstract manifests.